pym/bob/generators/QtCreatorGenerator.py

```diff
@@ -79,12 +81,16 @@ def addRunSteps(outFile, runTargets):
 
 def addBuildConfig(outFile, num, name, buildArgs, buildMeFile):
     outFile.write('<valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.' + str(num) + '">\n')
-    outFile.write('<value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">' + os.getcwd() + '</value>\n')
+    outFile.write('<value type="QString" key="ProjectExplorer.BuildConfiguration.BuildDirectory">' + (os.getcwd() if not isWindows() else os.popen('pwd -W').read().replace('\n', '')) + '</value>\n')
```

Comment:
we can't user getcwd for msys2, because of the generated qt-creator files shall be loaded under native windows qt-creator. So the used pathes have to be windows ones.

Reply:
I think we have to live with that. But I would suggest to move the "getcwd/pwd -W" dance into one location and pass the path from there to the functions.

Also I think you can replace the `.replace('\n', '')` with `.strip()` because I guess you just want to remove the trailing \n.